src/compiler/checker.ts

```diff
@@ -13917,13 +13917,15 @@ namespace ts {
                 if (propertiesArray.length > 0) {
                     spread = getSpreadType(spread, createObjectLiteralType());
                 }
-                if (spread.flags & TypeFlags.Object) {
-                    // only set the symbol and flags if this is a (fresh) object type
-                    spread.flags |= propagatedFlags;
-                    spread.flags |= TypeFlags.FreshLiteral;
-                    (spread as ObjectType).objectFlags |= ObjectFlags.ObjectLiteral;
-                    spread.symbol = node.symbol;
-                }
+                // only set the symbol and flags if this is a (fresh) object type
+                forEachType(spread, t => {
+                    if (t.flags & TypeFlags.Object) {
+                        t.flags |= propagatedFlags;
+                        t.flags |= TypeFlags.FreshLiteral;
+                        (t as ObjectType).objectFlags |= ObjectFlags.ObjectLiteral;
+                        t.symbol = node.symbol;

```

Comment:
Won't this trample over the existing symbol on any object types spread?

Reply:
Can we instead pass this symbol to getSpreadType and avoid the mutation after creating the type..

Reply:
@weswigham In its current state getSpreadType + the emptyObjectType check above will never trample an existing type. However, @mhegazy's idea is much less error-prone for future changes. Right now, getSpreadType is only called from places that have a symbol on hand anyway, since it's not a real type operator like it was planned to be.

Reply:
Done. It caused some churn in the ad-hoc JSX code, but it's probably worth it for the readability in checkObjectLiteral.